Fails to build with vs2026 and system code page 936

When building expat, random_rand_s.c reports that there's unexpected #endif on line 47.
The comment on line 46 contains the \u2018 and \u2019 mark, and It seems that the latest cl in vs2026 can't correctly handle this situation. After I removed the two marks, everything is ok.
This can be reproduced on windows system with Chinese locale, haven't tried other locales.

I ran cl and the output is like this:
```
Microsoft (R) C/C++ Optimizing Compiler Version 19.51.36248 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```
This issue is easy to be reproduced.
1. open command prompt for VS
2. run the following commands: 
```
powershell
Invoke-WebRequest https://github.com/libexpat/libexpat/releases/download/R_2_8_2/expat-2.8.2.tar.xz -Outfile expat-2.8.2.tar.xz
tar -xf .\expat-2.8.2.tar.xz
cd expat-2.8.2
cmake -G Ninja -B build -S .
cmake --build build
```
output is like this:
```
[1/38] Building C object CMakeFiles\expat.dir\lib\random_rand_s.c.obj
FAILED: [code=2] CMakeFiles/expat.dir/lib/random_rand_s.c.obj
C:\PROGRA~1\MICROS~2\18\COMMUN~1\VC\Tools\MSVC\1451~1.362\bin\Hostx64\x64\cl.exe  /nologo -DVER_FILEVERSION=2,8,2,0 -D_CRT_SECURE_NO_WARNINGS -Dexpat_EXPORTS -IC:\Users\zhw2101024\temp\expat-2.8.2\build -IC:\Users\zhw2101024\temp\expat-2.8.2\lib /DWIN32 /D_WINDOWS  /Zi /Ob0 /Od /RTC1 -MDd   -wd4996 /showIncludes /FoCMakeFiles\expat.dir\lib\random_rand_s.c.obj /FdCMakeFiles\expat.dir\ /FS -c C:\Users\zhw2101024\temp\expat-2.8.2\lib\random_rand_s.c
C:\Users\zhw2101024\temp\expat-2.8.2\lib\random_rand_s.c(1): warning C4819: The file contains a character that cannot be represented in the current code page (936). Save the file in Unicode format to prevent data loss
C:\Users\zhw2101024\temp\expat-2.8.2\lib\random_rand_s.c(47): fatal error C1020: unexpected #endif
[10/38] Building C object CMakeFiles\expat.dir\lib\xmltok.c.obj
ninja: build stopped: subcommand failed.
```
Further more, I tried to append one empty line after line 45 of the file random_rand_s.c, and this error disappeared, though there's other error messages and libexpat can be built after retries the last command several times. So i guess the problem is not on the unix file format itself, it's also related to the following `#if` statement.
Would you try this and see if it fails on your system?

@zhw2101024 I tried with your patch from #1288 on top now…

- commit 04b4324f8c66f4e2d1c56018f8a846d447d02442
- run https://github.com/libexpat/libexpat/actions/runs/29656775210/job/88112440639

…but the build still fails saying `random_rand_s.c(47,1): error C1020: unexpected #endif` and also shows warning `warning C4819: The file contains a character that cannot be represented in the current code page (936)` for two locations, e.g.:

- `tests\nsalloc_tests.c(1157,1)`
- `random_rand_s.c(1,1)`

I have a feeling that "codepage 936 mode" is trouble with compiling a codebase that is a mix of ASCII and UTF-8 source code. Is Expat the only project where you experience this class of error?

> [@zhw2101024](https://github.com/zhw2101024) I tried with your patch from [#1288](https://github.com/libexpat/libexpat/pull/1288) on top now…
> 
>     * commit [04b4324](https://github.com/libexpat/libexpat/commit/04b4324f8c66f4e2d1c56018f8a846d447d02442)
> 
>     * run https://github.com/libexpat/libexpat/actions/runs/29656775210/job/88112440639
> 
> 
> …but the build still fails saying `random_rand_s.c(47,1): error C1020: unexpected #endif` and also shows warning `warning C4819: The file contains a character that cannot be represented in the current code page (936)` for two locations, e.g.:
> 
>     * `tests\nsalloc_tests.c(1157,1)`
> 
>     * `random_rand_s.c(1,1)`
> 
The story is like this: after the first building failed, i made the change same with the path and rerun the cmake --build command and it passed, not sure what happend. The patch is definitely incorrect.

> I have a feeling that "codepage 936 mode" is trouble with compiling a codebase that is a mix of ASCII and UTF-8 source code. Is Expat the only project where you experience this class of error?

To tell the truth, i rarely compile code on windows. In the past i tried to build virtualbox and libreoffice and often fails with long error log, maybe all related the codepage. Maybe we can add the charset argument of cmake to the doc?

--- a/expat/CMakeLists.txt
+++ b/expat/CMakeLists.txt
@@ -206,7 +206,7 @@ if(MSVC)
     # - https://cmake.org/cmake/help/latest/variable/MSVC_VERSION.html
     # - https://sourceforge.net/p/predef/wiki/Compilers/
     # - https://en.wikipedia.org/wiki/Microsoft_Visual_Studio#History
-    set(_EXPAT_MSVC_REQUIRED_INT 1800)  # i.e. 12.0/2013/1800; see PR #426
+    set(_EXPAT_MSVC_REQUIRED_INT 1900)  # i.e. 14.0/2015(Update 2)/1900
     set(_EXPAT_MSVC_SUPPORTED_INT 1930)
     set(_EXPAT_MSVC_SUPPORTED_DISPLAY "Visual Studio 17.0/2022/${_EXPAT_MSVC_SUPPORTED_INT}")
 
@@ -219,6 +219,12 @@ if(MSVC)
             message(WARNING "Please use ${_EXPAT_MSVC_SUPPORTED_DISPLAY} or later.  Thank you!")
         endif()
     endif()
+
+    # Stop MSVC from trying to decode the source files using
+    # the Windows system code page encoding that could e.g. be Windows-936
+    # on a Chinese Windows system rather than UTF-8.
+    # https://devblogs.microsoft.com/cppblog/new-options-for-managing-character-sets-in-the-microsoft-cc-compiler/
+    add_compile_options(/source-charset:utf-8)
 endif()
 
 macro(_expat_copy_bool_int source_ref dest_ref)

